src/pages/EditUser/EditUser.tsx

```diff
@@ -0,0 +1,84 @@
+import { Container } from '@chakra-ui/react'
+import React from 'react'
+import { SubmitHandler } from 'react-hook-form'
+import { useMutation, useQuery, useQueryClient } from 'react-query'
+import { useNavigate, useParams } from 'react-router-dom'
+import { toast } from 'react-toastify'
+import { UserForm } from '../../components/layout/Form/UserForm'
+import { editUser, FormValues, getUserById, User } from '../../services/http/user'
+import { FormWrapper, Title, Wrapper } from './EditUser.styles'
+
+export const EditUser = () =>  {
+  const navigate = useNavigate()
+  const { id } = useParams()
+
+  const { data: user, isLoading, isError } = useQuery<User | null>(['user', id], () => getUserById(Number(id)), {
+    enabled: !!id,
+  })
+
+  const defaultFormValues: FormValues = {
+    name: '',
+    email: '',
+    profile: 'User',
+    age: null,
+    phone: '',
+    id: 0
+  }
+
+  if (user && !isLoading) {
+    defaultFormValues.name = user.name || defaultFormValues.name;
+    defaultFormValues.email = user.email || defaultFormValues.email;
+    defaultFormValues.profile = user.profile as 'Admin' | 'User' || defaultFormValues.profile;
+    defaultFormValues.age = user.age || defaultFormValues.age;
+    defaultFormValues.phone = user.phone || defaultFormValues.phone;
```

Comment:
Não devemos fazer mutações em variáveis criadas. Dificilmente isso deve ser feito e quando feito deve haver um motivo em especifico. A solução aqui é apenas enviar defaultValues para seu formulário. Se o valor do nome for undefined por exemplo o nome só não será exibido na tela... não é necessário setá-lo para '' como você está fazendo. Esse comentário vale para todos os locais onde você utiliza defaultValues dessa forma. Remova!